crates/rspack_core/src/dependencies_block.rs

```diff
@@ -0,0 +1,103 @@
+use std::sync::atomic::{AtomicU32, Ordering};
+
+use once_cell::sync::Lazy;
+use rspack_identifier::{Identifiable, Identifier};
+
+use crate::{BoxDependency, DependencyId, GroupOptions};
+
+pub trait DependenciesBlock {
+  fn add_block_id(&mut self, block: AsyncDependenciesBlockId);
+
+  fn get_blocks(&self) -> &[AsyncDependenciesBlockId];
+
+  fn add_dependency_id(&mut self, dependency: DependencyId);
+
+  fn get_dependencies(&self) -> &[DependencyId];
+}
+
+static ASYNC_DEPENDENCIES_BLOCK_ID: Lazy<AtomicU32> = Lazy::new(|| AtomicU32::new(0));
+
+fn get_async_dependencies_block_id() -> AsyncDependenciesBlockId {
+  AsyncDependenciesBlockId::from(
+    ASYNC_DEPENDENCIES_BLOCK_ID
+      .fetch_add(1, Ordering::Relaxed)
+      .to_string(),
+  )
+}
+
+pub type AsyncDependenciesBlockId = Identifier;
```

Comment:
Would be better if this is a new type, moving some runtime evaluation error to compile time:-)